Ripped from Debian

Patch to create a shared lib for libmp4ff.

--- a/common/mp4ff/Makefile.am
+++ b/common/mp4ff/Makefile.am
@@ -1,7 +1,8 @@
-lib_LIBRARIES = libmp4ff.a
+lib_LTLIBRARIES = libmp4ff.la
 include_HEADERS = mp4ff.h mp4ffint.h
 
-libmp4ff_a_CFLAGS = -DUSE_TAGGING=1
+AM_CFLAGS = -DUSE_TAGGING=1
 
-libmp4ff_a_SOURCES = mp4ff.c mp4atom.c mp4meta.c mp4sample.c mp4util.c \
+libmp4ff_la_LDFLAGS = -version-info 0:0:0 -no-undefined
+libmp4ff_la_SOURCES = mp4ff.c mp4atom.c mp4meta.c mp4sample.c mp4util.c \
 		     mp4tagupdate.c mp4ff.h mp4ffint.h mp4ff_int_types.h
--- a/frontend/Makefile.am
+++ b/frontend/Makefile.am
@@ -5,7 +5,7 @@
 	   -I$(top_srcdir)/common/mp4ff
 
 faad_LDADD = $(top_builddir)/libfaad/libfaad.la \
-	     $(top_builddir)/common/mp4ff/libmp4ff.a
+	     $(top_builddir)/common/mp4ff/libmp4ff.la
 
 faad_SOURCES = main.c \
 	       audio.c audio.h \
